pyccel/codegen/printing/pycode.py

```diff
@@ -34,9 +34,9 @@ class PythonCodePrinter(SympyPythonCodePrinter):
     ))
     _kc = {k: ''+v for k, v in _known_constants_math.items()}
 
-    def __init__(self, settings=None):
+    def __init__(self, parser=None, settings={}):
```

Comment:
The default is None to make codacy happy. See here for why : https://stackoverflow.com/a/41686973/5957195

Reply:
As a rule of thumb, default values should be immutable.